tests/infer/test_autoguide.py

```diff
@@ -449,12 +449,16 @@ def model():
 
 def test_unpack_latent():
     def model():
-        return pyro.sample('x', dist.LKJCorrCholesky(2, torch.tensor(1.)))
+        with pyro.plate('N', 5):
+            return pyro.sample('x', dist.LKJCorrCholesky(2, torch.tensor(1.)))
 
     guide = AutoDiagonalNormal(model)
     assert guide()['x'].shape == model().shape
     latent = guide.sample_latent()
-    assert list(guide._unpack_latent(latent))[0][1].shape == (1,)
+    assert list(guide._unpack_latent(latent))[0][1].shape == (5, 1)
+    with pyro.plate('N', 100):
```

Comment:
After looking at some of the failing autoguide tests in predictive, I was convinced that what we have is fine. To get this test to pass we will need to adjust this plate by `max_plate_dim` - `pyro.plate('N', 100, dim=-2)`. While `Predictive` does this by default, `MCMC.get_samples` does not consider `max_plate_dim`, so maybe we should change that so that those samples can be passed directly to other utilities like `guide._unpack_latent` without adjusting the sample dim.